server/konnectors/darty.js

```diff
@@ -0,0 +1,192 @@
+'use strict';
+
+const userAgent = 'Mozilla/5.0 (X11; Fedora; Linux x86_64; rv:37.0) Gecko/20100101 Firefox/37.0';
+const request = require('request').defaults({
+  jar: true,
+  headers: {
+    'User-Agent': userAgent
+  }
+});
+
+const moment = require('moment');
+const cheerio = require('cheerio');
+const async = require('async');
+const Pdf2json = require('pdf2json');
+const baseKonnector = require('../lib/base_konnector');
+const saveDataAndFile = require('../lib/save_data_and_file');
+const linkBankOperation = require('../lib/link_bank_operation');
+const filterExisting = require('../lib/filter_existing');
+const localization = require('../lib/localization_manager');
+const Bill = require('../models/bill');
+const log = require('printit')({
+  prefix: 'Darty',
+  date: true,
+});
+
+const fileOptions = {
+  vendor: 'Darty',
+  dateFormat: 'YYYYMMDD'
+};
+
+const host = 'https://secure.darty.com';
+const loginUrl = `${host}/espace_client/connexion`;
+const billsListUrl = `${host}/webapp/wcs/stores/controller/ec/orders`;
+
+function loginAndParseBillsList(fields, bills, data, next) {
+  request(loginUrl, (err) => {
+    if (err) {
+      return next('request error');
+    }
+    // Do nothing with the body, it is just to get an initial cookie
+    const options = {
+      method: 'POST',
+      url: loginUrl,
+      form: {
+        email: fields.login,
+        password: fields.password
+      }
+    };
+    request(options, (err, res) => {
+      if (err) {
+        return next(err);
+      }
+      // TODO : Improve test;
+      if (res.statusCode === 200) {
+        return next('bad credentials');
+      }
+      log.info('Connected');
+      request(billsListUrl, (err, res, body) => {
+        if (err) {
+          log.error(err);
+          return next('request error');
+        }
+        const $ = cheerio.load(body);
+        const billsList = [];
+
+        $('div.darty_sous_bloc_new_ec').each((i, element) => {
+          let billId = $('div.darty_sous_bloc_header_spacing_new_ec > span.red', element).text().trim();
+          billId = billId.replace(/\r\n/g, '').replace(/ /g, '').replace(/\/[A-Z]+/, '');
+
+          const date = $('div > span.label > strong.red', element).text();
+          const pdfurl = $('div.parc_facture_cell_new_ec > a.darty_sous_bloc_action_link_new_ec', element).attr('href');
+          // Not all the orders on the website have a bill
+          if (typeof pdfurl !== 'undefined') {
+            billsList.push({
+              number: billId,
```

Comment:
What's the purpose of this `number` field ? Does it expect an number type or simply a number-ish thing ?